node/src/blockchain/blockchain_interface.rs

```diff
@@ -384,10 +412,58 @@ where
 
 impl<T> BlockchainInterfaceNonClandestine<T>
 where
-    T: Transport + Debug + 'static,
+    T: 'static + BatchTransport + Debug,
+{
+    fn find_largest_transaction_block_number(
+        &self,
+        response_block_number: u64,
+        transactions: &Vec<BlockchainTransaction>,
+    ) -> u64 {
+        if transactions.is_empty() {
+            response_block_number
+        } else {
+            transactions
+                .iter()
+                .fold(response_block_number, |so_far, elem| {
+                    if elem.block_number > so_far {
+                        elem.block_number
+                    } else {
+                        so_far
+                    }
+                })
+        }
+    }
+}
+
+impl<T> BlockchainInterfaceNonClandestine<T>
+where
+    T: 'static + BatchTransport + Debug,
+{
+    fn extract_transactions_from_logs(&self, logs: Vec<Log>) -> Vec<BlockchainTransaction> {
+        logs.iter()
+            .filter_map(|log: &Log| match log.block_number {
+                None => None,
+                Some(block_number) => {
+                    to_gwei(U256::from(log.data.0.as_slice())).map(|gwei_amount| {
+                        BlockchainTransaction {
+                            block_number: block_number.as_u64(),
+                            from: Wallet::from(log.topics[1]),
+                            gwei_amount,
```

Comment:
One important thing here, by the time of this writing we've got already "in-wei counting" system implemented. That means that the other Nodes will probably pay you something different than just clear gwei values. These numbers moved between Nodes are going to be much more accurate. So you don't want to lose the accuracy here by conversion to gwei. You'll have to change the field's name and the data type to u128, plus get rid of the conversion. 

It might be the case that it will be automatically taken care of by doing the merge (I might have adjusted this code already...well, given your method is now concise and placed alone...the merge might not affect this code...but its mostly the same so maybe yes.). Please check it out anyway.